fab/utils/prioritised_replay_buffer.py

```diff
@@ -116,11 +116,17 @@ def adjust(self, log_w_adjustment, log_q, indices):
         """Adjust log weights and log q to match new value of theta, this is typically performed
         over minibatches, rather than over the whole dataset at once."""
         valid_adjustment = ~torch.isinf(log_w_adjustment) & ~torch.isnan(log_q)
-        log_w_adjustment, log_q, indices = \
+        log_w_adjustment, log_q, valid_indices = \
             log_w_adjustment[valid_adjustment], log_q[valid_adjustment], indices[valid_adjustment]
-        indices = indices.to(self.device)
-        self.buffer.log_w[indices] += log_w_adjustment.to(self.device)
-        self.buffer.log_q_old[indices] = log_q.to(self.device)
+        valid_indices = valid_indices.to(self.device)
+        self.buffer.log_w[valid_indices] += log_w_adjustment.to(self.device)
+        self.buffer.log_q_old[valid_indices] = log_q.to(self.device)
+
+        # Kill samples in the buffer for which the `log_w_adjustment` is invalid.
+        # A common reason this can occur is if AIS discovers a point far outside the reasonable range of the problem.
+        # Which causes nan log probs under the flow.
+        invalid_indices = indices[~valid_adjustment]
+        self.buffer.log_w[invalid_indices] = -torch.ones_like(self.buffer.log_w[invalid_indices])*(float("inf"))
```

Comment:
Prevent invalid samples from being sampled again by setting their importance weight to -inf in the buffer. 
One example of when this can occur is in the Many Well problem, HMC can sometimes return samples way outside the reasonable range that we care about (e.g. 100) causing the flow to return NaN. We remove these samples as they are un-usable for training (cause the loss to be NaN).